daemon/algod/api/algod.oas2.json

```diff
@@ -4127,6 +4144,52 @@
         }
       }
     },
+    "ApplicationKVStorage": {
```

Comment:
We have some types for existing responses that it would be nice to use. For example, there is
```
    "Application": {
      "description": "Application index and its parameters",
```
That would be nice for returning global information about apps.
Inside, it uses
```
        "global-state": {
          "description": "\\[gs\\] global state",
          "$ref": "#/definitions/TealKeyValueStore"
        }
```
which seems like it ought to work for this purpose too.

Similarly, we have types that are used when someone uses the `/accounts/{address}/applications/{application-id}"` endpoint which can return local states, and more types for assets under `/assets/` for asset params and holdings. And another for returning boxes from `"/v2/applications/{application-id}/boxes"`


Can we re-use them?

Reply:
Good point, I was a bit hesitating using `Application`, for it somehow contradicts the purpose of introducing `AVMValue` at the first place -- in side of `Application` uses `TEALValue` which is the one we want to replace... But structurally `Application` object looks better, and if this is more preferred I can revert to use `Application` object and keep the object change minimal.

Reply:
Can you remind me of the issue with TEALValue? I think it had something to do with encoding bytes?  It feels like we're recreating an awful lot stuff that's nearly identical and I'd really like to avoid it.

Reply:
right, the introduction of `AVMValue` in oas2 is that, `TEALValue` require both uint and byte fields to be required, and bytes are not formated with byte (b64 encode), just string.

I think under this case we probably be more prone to reuse the `Application` object then, for eventually we will write something for account and assets, and we don't want to rewrite another batch of oas2 objects (though I really want to see if we can deprecate the `TEALValue` object and use `AVMValue` as substitution...)

Reply:
Even setting aside the `TEALValue` vs `AVMValue` issue, would it even make sense to use `Application`, given that that definition includes app param information beyond just the global state?

If this PR were attempting to return the entire record of an account/asset/app, I think I would agree to use the preexisting types, but since it's intentionally targetting a smaller set of information, it seems like we need a new definition anyway.

Reply:
Like the Schema and Program? I guess we don't have to return them, but I suspect we will want to _accept_ them, in the style of dry run, and it would be nice to use the same type.

For assets, I think we do want to return asset params, as they are visible in AVM. (I suppose program and schema are too, actually, and I guess it makes sense to see them too. But I admit the value is not high.)

Reply:
If the extra types seems like no big deal, I won't argue too hard.

Reply:
I guess my decision hinges on whether this PR is more of a one-off attempt to help expose app state values, or whether it's the start of exposing all possible initial state (e.g. app params, asset params/holdings, account data).

If the latter, then `Application` and other preexisting types make more sense (and I'd probably argue that a separate flag is needed).

Reply:
I suppose we can live we exposing globals/locals/boxes.  I was thinking we'd want to expose other things that can change during txn execution like asset params and balances/holdings. But they are a lot further down in priority, since they are not going to be changed willy-nilly by opcodes executing.  Only by actual transactions. And so a debugger might be sad it can't show you an asset's params, as soon as your code accesses them you will see them on the stack.